modules/openapi-generator/src/main/resources/typescript-axios/apiInner.mustache

```diff
@@ -203,7 +203,8 @@ export const {{classname}}AxiosParamCreator = function (configuration?: Configur
             localVarRequestOptions.data = localVarFormParams{{#vendorExtensions}}{{^multipartFormData}}.toString(){{/multipartFormData}}{{/vendorExtensions}};
     {{/hasFormParams}}
     {{#bodyParam}}
-            const needsSerialization = (typeof {{paramName}} !== "string") || localVarRequestOptions.headers['Content-Type'] === 'application/json';
+            const contentType = localVarRequestOptions.headers['Content-Type'];
+            const needsSerialization = (typeof {{paramName}} !== "string") && (contentType.match(JSON_MIME_PATTERN) || contentType.match(JSON_VENDOR_MIME_PATTERN));
```

Comment:
This could be a breaking change if people were relying on this `JSON.stringify` with either an empty or non-JSON `Content-Type`. However, I didn't want to just add an exclusion for `application/octet-stream` since others, like `image/jpeg`, etc. wouldn't want `JSON.stringify` called either.